go/teams/seitan_v2.go

```diff
@@ -0,0 +1,226 @@
+
+	keyPair, err := sikey.generateKeyPair()
+	if err != nil {
+		return pkey, encoded, err
+	}
+
+	var keyAndLabel keybase1.SeitanKeyAndLabelVersion2
+	keyAndLabel.K = keybase1.SeitanPubKey(keyPair.GetKID().String())
+	keyAndLabel.L = label
+
+	packedKeyAndLabel, err := libkb.MsgpackEncode(keybase1.NewSeitanKeyAndLabelWithV2(keyAndLabel))
+	if err != nil {
+		return pkey, encoded, err
+	}
+	return packAndEncryptKeyWithSecretKey(secretKey, gen, nonce, packedKeyAndLabel, SeitanVersion2)
+}
+
+func (sikey SeitanSIKeyV2) GeneratePackedEncryptedKey(ctx context.Context, team *Team, label keybase1.SeitanKeyLabel) (pkey SeitanPKey, encoded string, err error) {
+	appKey, err := team.SeitanInviteTokenKeyLatest(ctx)
+	if err != nil {
+		return pkey, encoded, err
+	}
+
+	var nonce keybase1.BoxNonce
+	if _, err = rand.Read(nonce[:]); err != nil {
+		return pkey, encoded, err
+	}
+
+	return sikey.generatePackedEncryptedKeyWithSecretKey(appKey.Key, appKey.KeyGeneration, nonce, label)
+}
+
+// "Signature"
+type SeitanSig libkb.NaclSignature
+type SeitanPubKey libkb.NaclSigningKeyPublic
+
+func GenerateSeitanSignatureMessage(uid keybase1.UID, eldestSeqno keybase1.Seqno, inviteID SCTeamInviteID, time keybase1.Time) (payload []byte, err error) {
+	type SigPayload struct {
+		Stage       string         `codec:"stage" json:"stage"`
+		UID         keybase1.UID   `codec:"uid" json:"uid"`
+		EldestSeqno keybase1.Seqno `codec:"eldest_seqno" json:"eldest_seqno"`
+		CTime       keybase1.Time  `codec:"ctime" json:"ctime"`
+		InviteID    SCTeamInviteID `codec:"invite_id" json:"invite_id"`
+		Version     SeitanVersion  `codec:"version" json:"version"`
+	}
+
+	payload, err = libkb.MsgpackEncode(SigPayload{
+		Stage:       "accept",
+		Version:     SeitanVersion2,
+		InviteID:    inviteID,
+		UID:         uid,
+		EldestSeqno: eldestSeqno,
+		CTime:       time,
+	})
+	return payload, err
+}
+
+func VerifySeitanSignatureMessage(pubKey SeitanPubKey, msg []byte, sig SeitanSig) error {
+	naclsig := libkb.NaclSignature(sig)
+	valid := libkb.NaclSigningKeyPublic(pubKey).Verify(msg, &naclsig)
+	if !valid {
+		return libkb.KeyCannotVerifyError{}
+	}
+	return nil
+}
+
+func ImportSeitanPubKey(keyString keybase1.SeitanPubKey) (pubKey SeitanPubKey, err error) {
+	keypair, err := libkb.ImportNaclSigningKeyPairFromHex(string(keyString))
+	if err != nil {
+		return pubKey, err
+	}
+	return SeitanPubKey(keypair.Public), nil
+}
+
+func (sikey SeitanSIKeyV2) GenerateSignature(uid keybase1.UID, eldestSeqno keybase1.Seqno, inviteID SCTeamInviteID, time keybase1.Time) (sig SeitanSig, encoded string, err error) {
+	payload, err := GenerateSeitanSignatureMessage(uid, eldestSeqno, inviteID, time)
+	if err != nil {
+		return sig, encoded, err
+	}
+
+	keyPair, err := sikey.generateKeyPair()
+	if err != nil {
+		return sig, encoded, err
+	}
+
+	sig = SeitanSig(*keyPair.Private.Sign(payload))
```

Comment:
What does the output of this look like?  Is it a raw sig or a `NaclSigInfo`?  I guess I was expecting a `NaclSigInfo` rather than a raw sig.

Reply:
I could be OK with a raw sig....

Reply:
It's just a raw sig right now -- I was trying to find examples in the code base for which methods to use and somehow settled on the raw sig when I first implemented. I can refactor or leave as is, which do you prefer?